hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightSearchEndpoint.java

```diff
@@ -0,0 +1,427 @@
+import static org.apache.hadoop.ozone.recon.api.handlers.EntityHandler.normalizePath;
+import static org.apache.hadoop.ozone.recon.api.handlers.EntityHandler.parseRequestPath;
+
+/**
+ * REST endpoint for search implementation in OM DB Insight.
+ */
+@Path("/insights")
+@Produces(MediaType.APPLICATION_JSON)
+@AdminOnly
+public class OMDBInsightSearchEndpoint {
+
+  private OzoneStorageContainerManager reconSCM;
+  private final ReconOMMetadataManager omMetadataManager;
+  private static final Logger LOG =
+      LoggerFactory.getLogger(OMDBInsightSearchEndpoint.class);
+  private ReconNamespaceSummaryManagerImpl reconNamespaceSummaryManager;
+
+
+  @Inject
+  public OMDBInsightSearchEndpoint(OzoneStorageContainerManager reconSCM,
+                                   ReconOMMetadataManager omMetadataManager,
+                                   ReconNamespaceSummaryManagerImpl reconNamespaceSummaryManager) {
+    this.reconSCM = reconSCM;
+    this.omMetadataManager = omMetadataManager;
+    this.reconNamespaceSummaryManager = reconNamespaceSummaryManager;
+  }
+
+
+  /**
+   * Performs a search for open keys in the Ozone Manager (OM) database using a specified search prefix.
+   * This endpoint can search across both File System Optimized (FSO) and Object Store (non-FSO) layouts,
+   * compiling a list of keys that match the given prefix along with their data sizes.
+   *
+   * The search prefix may range from the root level ('/') to any specific directory
+   * or key level (e.g., '/volA/' for everything under 'volA'). The search operation matches
+   * the prefix against the start of keys' names within the OM DB.
+   *
+   * Example Usage:
+   * 1. A startPrefix of "/" will return all keys in the database.
+   * 2. A startPrefix of "/volA/" retrieves every key under volume 'volA'.
+   * 3. Specifying "/volA/bucketA/dir1" focuses the search within 'dir1' inside 'bucketA' of 'volA'.
+   *
+   * @param startPrefix   The prefix for searching keys, starting from the root ('/') or any specific path.
+   * @param includeFso    Indicates whether to include FSO layout keys in the search.
+   * @param includeNonFso Indicates whether to include non-FSO layout keys in the search.
+   * @param limit         Limits the number of returned keys.
+   * @return A KeyInsightInfoResponse, containing matching keys and their data sizes.
+   * @throws IOException On failure to access the OM database or process the operation.
+   */
+  @GET
+  @Path("/openKeys/search")
+  public Response searchOpenKeys(
+      @DefaultValue(DEFAULT_SEARCH_PREFIX) @QueryParam("startPrefix")
+      String startPrefix,
+      @DefaultValue(DEFAULT_OPEN_KEY_INCLUDE_FSO) @QueryParam("includeFso")
+      boolean includeFso,
+      @DefaultValue(DEFAULT_OPEN_KEY_INCLUDE_NON_FSO) @QueryParam("includeNonFso")
+      boolean includeNonFso,
+      @DefaultValue(RECON_OPEN_KEY_SEARCH_LIMIT) @QueryParam("limit")
+      int limit) throws IOException {
+    if (limit < 0) {
```

Comment:
IMO, lets try to get rid of this validation and take default as `Zero` if user pass negative value of limit.

Reply:
Done default zero set to limit for -ve values.